code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/DataQueueService.java

```diff
@@ -53,11 +49,16 @@ public DataQueue getDataQueue(final String databaseName) {
 				dataQueue = dataQueueCache.get(databaseName);
 
 				if (dataQueue == null) {
-					final File databaseDirDataQueue = openOrMigrateExistingDataQueue(databaseName);
+					final File databaseDirDataQueue = FileUtil.openOrCreateDatabase(
```

Comment:
Let us not add db specific methods to FileUtil. I would just recommend adding a `move()` to move the db from old location to new location. 

I would recommend this init logic to be 
```
File db_path = construct_new_db_path
if (!db_path.exists()) {
  // Try migration from cache dir 
  File cache_dir_db_path = construct_old_db_path
  FileUtil.move( cache_dir_db_path, db_path);
}

// If it still does not exist migration failed. Create a new file.
if (!db_path.exists()) {
   db_path.createFile();
}
```